tests/libvmi/storage.go

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package libvmi
+
+import (
+	kvirtv1 "kubevirt.io/client-go/api/v1"
+)
+
+// WithContainerImage specifies the name of the container image to be used.
+func WithContainerImage(name string) Option {
+	return func(vmi *kvirtv1.VirtualMachineInstance) {
+		diskName, bus := "disk0", "virtio"
+		vmi.Spec.Domain.Devices.Disks = append(vmi.Spec.Domain.Devices.Disks, newDisk(diskName, bus))
+		vmi.Spec.Volumes = append(vmi.Spec.Volumes, newContainerVolume(diskName, name))
+	}
+}
+
+func addDisk(vmi *kvirtv1.VirtualMachineInstance, disk kvirtv1.Disk) {
```

Comment:
So many helpers just for storage. And it is not even hiding too much logic, just the structure.

Feels too low of an abstraction to be worth so many functions. For somebody who knows basics of VMI API it would be easier to read the structure alone. Jumping between these functions is tiresome.

Don't you think we should put the bar higher, don't put these simple assignments to functions and only abstract higher level logic?